echopop/inversion/inversion_matrix.py

```diff
@@ -0,0 +1,1230 @@
+import time
+import warnings
+from typing import Any, Dict, Literal, Optional
+
+import numpy as np
+import pandas as pd
+from lmfit import Minimizer
+from pydantic import ValidationError
+
+from ..core.exceptions import EchopopValidationError
+from ..inversion.inversion_base import InversionBase
+from ..typing import InvParameters, MCInvParameters
+from ..validators.inversion import (
+    ValidateBuildModelArgs,
+    ValidateInversionMatrix,
+)
+from ..validators.scattering_models import SCATTERING_MODEL_PARAMETERS
+
+
+def echopop_optim_cb(params, iter, resid, *args, **kwargs):
```

Comment:
suggest function name change to make it clear what the function does: `minimizer_print_callback`.